core/sds.c

```diff
@@ -1828,15 +1854,17 @@ _upload_sequential (struct oio_sds_s *sds, struct oio_sds_ul_dst_s *dst,
 
 	oio_pfree0(&b, NULL);
 
-	if (!err)
+	if (!err) {
 		err = oio_sds_upload_commit (ul);
-	else {
-		struct oio_error_s *e = oio_sds_upload_abort (ul);
-		if (e) {
-			GRID_WARN("Upload abort failed: (%d) %s",
-					oio_error_code (e), oio_error_message (e));
-			oio_error_free (e);
+		if (oio_error_code(err) == CODE_CONTENT_EXISTS ||
+				oio_error_code(err) == CODE_CONTENT_PRECONDITION) {
+			_upload_abort_no_error(ul);
```

Comment:
Since oio-sds 4.1.9, both `CODE_CONTENT_EXISTS` and `CODE_CONTENT_PRECONDITION` are converted to `HTTP_CODE_CONFLICT` by the proxy, thus we must only check for this one.

Reply:
Sadly, the C proxy client (at core/proxy.c +395) doesn't really take care of the HTTP response code and always tries to parse the body as a JSON error (object with "status" and "message").